File: src/main/java/hacker/solution/DequeDemo.java
package hacker.solution;

import java.util.*;

public class DequeDemo {
        public static void main(String[] args) {
            Scanner in = new Scanner(System.in);
            Deque deque = new ArrayDeque<>();
            int n = in.nextInt();
            int m = in.nextInt();
            int max = 0;
            Set hs = new HashSet<>();
            for (int i = 0; i < n; i++) {
                int num = in.nextInt();
                deque.add(num);
                hs.add(num);
                if (deque.size() == m+1) {
                    int removed = (int)deque.remove();
                    if (!deque.contains(removed)) {
                        hs.remove(removed);
                    }
                }
                max = Math.max(hs.size(),max);
            }
            System.out.println(max);
        }
}

File: src/main/kotlin/hacker/solutionk/DequeDemo.kt
package hacker.solutionk

import java.util.*

object DequeDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        val `in` = Scanner(System.`in`)
        val deque = ArrayDeque<Any>()
        val n = `in`.nextInt()
        val m = `in`.nextInt()
        var max = 0
        val hs = HashSet<Any>()
        for (i in 0 until n) {
            val num = `in`.nextInt()
            deque.add(num)
            hs.add(num)
            if (deque.size == m + 1) {
                val removed = deque.remove() as Int
                if (!deque.contains(removed)) {
                    hs.remove(removed)
                }
            }
            max = Math.max(hs.size, max)
        }
        println(max)
    }
}